src/Sensors/I2C/Loom_MultiGasSensor/Loom_MultiGasSensor.cpp

```diff
@@ -0,0 +1,148 @@
+#include "Loom_MultiGasSensor.h"
+#include "Wire.h"
+#include "Logger.h"
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////
+Loom_MultiGasSensor::Loom_MultiGasSensor(
+                            Manager& man,
+                            uint8_t address,
+                            bool useMux
+                    ) : I2CDevice("MultiGasSensor"), manInst(&man), addr(address), gas(&Wire, address) {
+                        module_address = address;
+
+                        // Register the module with the manager
+                        if(!useMux)
+                            manInst->registerModule(this);
+                    }
+//////////////////////////////////////////////////////////////////////////////////////////////////////
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////
+uint8_t Loom_MultiGasSensor::get_gas_i2c(const std::string& gasType) {
+    static const std::unordered_map<std::string, uint8_t> GAS_ADDRESSES = {
```

Comment:
We make a new one of these every time, but it seems like it doesn't change, move to a member variable